src/global/sync.rs

```diff
@@ -0,0 +1,488 @@
+    {
+        let mut script = activation_script.clone();
+        shell
+            .run_command(
+                &mut script,
+                [
+                    format!(
+                        "\"{}\"",
+                        prefix.root().join(original_executable).to_string_lossy()
+                    )
+                    .as_str(),
+                    get_catch_all_arg(shell),
+                ],
+            )
+            .expect("should never fail");
+
+        if matches!(shell, ShellEnum::CmdExe(_)) {
+            // wrap the script contents in `@echo off` and `setlocal` to prevent echoing the
+            // script and to prevent leaking environment variables into the
+            // parent shell (e.g. PATH would grow longer and longer)
+            script = format!("@echo off\nsetlocal\n{}\nendlocal", script);
+        }
+
+        tokio::fs::write(&global_script_path, script)
+            .await
+            .into_diagnostic()?;
+
+        #[cfg(unix)]
+        {
+            use std::os::unix::fs::PermissionsExt;
+            std::fs::set_permissions(global_script_path, std::fs::Permissions::from_mode(0o755))
+                .into_diagnostic()?;
+        }
+    }
+    Ok(())
+}
+
+/// Warn user on dangerous package installations, interactive yes no prompt
+pub(crate) fn prompt_user_to_continue(
+    packages: &IndexMap<PackageName, MatchSpec>,
+) -> miette::Result<bool> {
+    let dangerous_packages = HashMap::from([
+        ("pixi", "Installing `pixi` globally doesn't work as expected.\nUse `pixi self-update` to update pixi and `pixi self-update --version x.y.z` for a specific version."),
+        ("pip", "Installing `pip` with `pixi global` won't make pip-installed packages globally available.\nInstead, use a pixi project and add PyPI packages with `pixi add --pypi`, which is recommended. Alternatively, `pixi add pip` and use it within the project.")
+    ]);
+
+    // Check if any of the packages are dangerous, and prompt the user to ask if
+    // they want to continue, including the advice.
+    for (name, _spec) in packages {
+        if let Some(advice) = dangerous_packages.get(&name.as_normalized()) {
+            let prompt = format!(
+                "{}\nDo you want to continue?",
+                console::style(advice).yellow()
+            );
+            if !dialoguer::Confirm::new()
+                .with_prompt(prompt)
+                .default(false)
+                .show_default(true)
+                .interact()
+                .into_diagnostic()?
+            {
+                return Ok(false);
+            }
+        }
+    }
+
+    Ok(true)
+}
+
+pub(crate) async fn sync(
+    env_root: EnvRoot,
+    project: super::Project,
+    bin_dir: BinDir,
+    config: Config,
+    gateway: rattler_repodata_gateway::Gateway,
+    auth_client: reqwest_middleware::ClientWithMiddleware,
+) -> Result<(), miette::Error> {
+    // Prune environments that are not listed
+    env_root
```

Comment:
we need a warning here at least until it's in development stage because it is performing a destructive operation